frontend/utils/auth.js

```diff
@@ -23,28 +25,37 @@ const ID_TOKEN_KEY = 'id_token';
 const ACCESS_TOKEN_KEY = 'access_token';
 const TOKEN_EXPIRE = 'token_expire';
 const USER_NAME = 'user_name';
+const USER_EMAIL = 'user_email';
 const PROFILE_THUMBNAIL = 'profile_thumbnail';
 
+const FB_SCOPE = 'public_profile, email';
+
 var router = new Router({
    mode: 'history',
 });
 
 export function login() {
   FB.getLoginStatus(function(response) {
     statusChangeCallback(response);
-  }, {scope: 'public_profile'});
+  }, {scope: FB_SCOPE});
 }
 
 function statusChangeCallback(response) {
-  console.log("statusChangeCallback");
   if (response.status == 'connected') {
     setTokens(response.authResponse);
-    FB.api('/me', {fields: 'name,picture'}, function(response) {
+    FB.api('/me', {fields: 'name,picture,email'}, function(response) {
       configureUser(response);
+      const userInfo = {
+        id: localStorage[ID_TOKEN_KEY],
+        email: localStorage[USER_EMAIL],
+        name: localStorage[USER_NAME],
+        picture: localStorage[PROFILE_THUMBNAIL],
+      };
+      api.updateUser(userInfo);
```

Comment:
Do you want to make it so that the reroute happens after the `updateUser` call using a `.then(response) {}`?